### PR TITLE
Logger: Add feature toggle for errors in HTTP request logs

### LINKED ISSUE #64378: Make request logs wide to include information about errors from errutil
Combine the error logger and request logger so that we log one wide message rather than multiple rows when an error occurs:

> The lack of error message in the route logger is very annoying :)
> Might be wise to combine these log statement into one. Logging an enriched log line for every http request is very reasonable for a web application.

from @bergquist

### PR DESCRIPTION
**Pull request by sakjur**
**What is this feature?**

Introduce a new flag for `unifiedRequestLog` that puts errors coming from any source on the HTTP request log line rather than on a separate line.

**Why do we need this feature?**

Moves us toward having "one line per request", which is a generally nice property to debug quickly.

**Who is this feature for?**

Grafana system administrators and developers.

**Which issue(s) does this PR fix?**:

Fixes #64378 

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/middleware/loggermw/logger.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+	var gfErr errutil.Error
+	if !errors.As(err, &gfErr) {
+		return []any{"err", err.Error()}
+	}
+
+	return []any{
+		"errReason", gfErr.Reason,
+		"errMessageID", gfErr.MessageID,
+		"err", gfErr.LogMessage,
```
Comment (marefr) [inline pkg/middleware/loggermw/logger.go:148]: Nit. Should we use `err` or `error`? We used `error` in response.go earlier. Think both being used throughout the code base.

**Review by marefr — COMMENTED** [review]
Looks good I think. Added some comments,

File: pkg/services/contexthandler/model/model.go [hunk 2]
```diff
@@ -138,6 +139,11 @@ func (ctx *ReqContext) writeErrOrFallback(status int, message string, err error)
 			}
 		}
 
+		if errutil.HasUnifiedLogging(ctx.Req.Context()) {
+			ctx.Error = err
+			logger = log.NewNopLogger().Info
+		}
```
[the remark this answers is not in the document]
Comment (sakjur) [reply pkg/services/contexthandler/model/model.go:145]: Good point, this is the result of a refactoring without a second pass over the logic.

**Review by bergquist — APPROVED** [review]
LGTM

**Review by chri2547 — APPROVED** [review]
Docs LGTM. Thank you for the contribution.

File: pkg/api/response/response.go [hunk 3]
```diff
@@ -80,21 +80,11 @@ func (r *NormalResponse) ErrMessage() string {
...
 	if r.err != nil {
-		v := map[string]interface{}{}
-		traceID := tracing.TraceIDFromContext(ctx.Req.Context(), false)
-		if err := json.Unmarshal(r.body.Bytes(), &v); err == nil {
-			v["traceID"] = traceID
-			if b, err := json.Marshal(v); err == nil {
-				r.body = bytes.NewBuffer(b)
-			}
+		if errutil.HasUnifiedLogging(ctx.Req.Context()) {
+			ctx.Error = r.err
```
Comment (suntala) [inline pkg/api/response/response.go:84]: I'm still fuzzy on parts of this PR. Is the idea that errors will only be captured by `ctx.Error` and that `ctx.Logger` will be used for other types of logs?

Comment (sakjur) [reply pkg/api/response/response.go:84]: Yeah, basically.

Not trying to fix anything beyond the somewhat standardized error logging at the moment.

**Pull request merged**